checker/examples/Makefile

```diff
@@ -4,7 +4,6 @@ all: compile
 
 compile:
 	$(JAVAC) *.java
-	$(JAVAC) units-extension/*.java
```

Comment:
I personally usually think of the top-level makefile as doing everything at its level and below (perhaps delegating for subdirectories).  That seems easier for people to use, and less error-prone because users are less lilkely to forget some parts.

If you want to do it the other way, then I suggest this be clearly documented and there should still be a single target somewhere that does all the work.